### PR TITLE
Added cross entropy and learning rate scheduler support to validation training, simplified metric reporting

### LINKED ISSUE #4807: ImageClassifier with ExponentialLRDecay: metrics not updated/calculated during validation
### System information

- **OS version**: Windows 10 Pro 18363
- **.NET Version**:  Core 2.1
- **Platform**: x64
- **ML.NET version**: 0.15-preview

### Issue

- **What did you do?**
  I am trying to train an image classifier that makes use of `ExponentialLRDecay`.
  I would like to see the metrics for training and validation for each epoch.

  ```csharp
  var options = new ImageClassificationTrainer.Options()
  {
    LearningRateScheduler = new ExponentialLRDecay(),
    ValidationSetFraction = 0.1f,
    MetricsCallback = (metrics) => Console.WriteLine(metrics  + $"   CrossEntropy: {metrics.Train.CrossEntropy}, LearningRate: {metrics.Train.LearningRate})
  };
  var model = mlContext.MulticlassClassification.Trainers.ImageClassification(options);
  ```

  (I added `  + $"   CrossEntropy: {metrics.Train.CrossEntropy}, LearningRate: {metrics.Train.LearningRate}"`, because the crossentropy and learning rate are not printed by default for the validation set.)

- **What happened?**
  * The learning rate is not updated for the validation set (seen on every even row in the image).
  * The cross-entropy is not calculated for the validation set (seen on every even row in the image).
  * The learning rate is not updated after the second epoch like the default value of `2` for `numEpochsPerDecay ` in `ExponentialLRDecay()`, but after the first instead (seen on the third row in the image). After that, the learning rate is correctly updated every 2 epochs. I'm not sure if this is the expected behavior.

  image

- **What did you expect?**
  * I expected a decaying learning rate in the validation step, equal to the one in the training step.
  * I expected the cross-entropy to be calculated in the validation step. The model with the highest `Accuracy` and lowest `CrossEntropy` are the best, so if 2 models perform equally well in terms of accuracy, the one with the lowest cross-entropy on the validation set should be picked.
  * Further, I expected the learning rate to start decaying after the 2nd epoch.

### PR DESCRIPTION
**Pull request by mstfbl**
Fix #4807 

In this PR, I've added cross entropy metric calculation & reporting during validation training in the Image Classification Trainer. Now, validation training can obtain cross entropy values for each epoch during validation training.

I have attached below screenshots of training and validation metrics with this [repro](https://github.com/dotnet/machinelearning-samples/compare/master...gartangh:validation_metrics).

Before (note that learning rate is constant, even when the given `ExponentialLRDecay` is used, and cross entropy during validation is reported as 0):
before_2

After:
_after

I have also edited `ToString()` to report accuracy and cross entropy for the validation phase as well. Reporting learning rate decay during validation isn't necessary, as no learning rate decay occurs in validation.

I have also added tests to verify the proper decaying of the cross-entropy rate with this PR.

### CODE REVIEW

**Comment by mstfbl** [comment]
Found a small bug in my implementation, will mark PR for review after fix.

**Comment by mstfbl** [comment]
Hey Ye,
Thank you so much for your review. After checking in with @harshithapv, we've arrived at the concolusion that the validation stage does not require learning rate scheduler support, as validation is only for computing loss and analyzing how the training phase is going. As there is no actual model training during the validation phase, there is also no need for decaying learning rates during validation. 
I'll be making the necessary changes soon!

File: src/Microsoft.ML.Vision/ImageClassificationTrainer.cs [hunk 1]
```diff
@@ -170,11 +170,11 @@ public sealed class TrainMetrics
             public override string ToString()
             {
                 if (DatasetUsed == ImageClassificationMetrics.Dataset.Train)
-                    return $"Phase: Training, Dataset used: {DatasetUsed.ToString(),10}, Batch Processed Count: {BatchProcessedCount,3}, Learning Rate: {LearningRate,10} " +
```
Comment (mstfbl) [inline src/Microsoft.ML.Vision/ImageClassificationTrainer.cs:173]: I have added learning rate to the end of the string here for cosmetic reasons. Now there is uniform alignment of columns' statistics between training and validation. :)

**Comment by mstfbl** [comment]
@wangyems I have removed learning rate decay for validation from my PR, and now the accuracy statistics for before and after are the same. Now cross entropy is being reported for validation, and as it should be, there is no change in learning rates during the validation phase,
